react/src/lib/components/EdsIcon/EdsIcon.jsx

```diff
@@ -0,0 +1,13 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { EdsIcon as EdsIconComponent } from "./components/EdsIcon";
+
+export const EdsIcon = (props) => {
+    return <EdsIconComponent {...props} />;
+};
+
+EdsIcon.propTypes = {
+    icon: PropTypes.string.isRequired,
+    size: PropTypes.oneOf([16, 24, 32, 40, 48]),
+    color: PropTypes.string,
+};
```

Comment:
so how come you have to define the propTypes twice? and why does one need this very thin `jsx` wrapper around the `tsx` file? I'm sorry, I'm rather new to typescript...

Reply:
This is in order to make `dash-generate-components` recognize the component when creating the Python wrapper around it. With the newer versions of Dash it should be able to recognize TypeScript, but there are still some issues with the newest Dash version so we cannot use it yet.

Reply:
As you can see, we followed this pattern for all components in `wcc`. By the time the new solution is in place, we are going to adjust the whole package (including operations in the `package.json`). If you wonder why we're defining TypeScript and PropTypes, they are not doing exactly the same. TypeScript is checked at build time while PropTypes are checked at runtime. Since we don't know the exact context in which the components are going to be used (might be in e.g. an API with types coming in at runtime), we are equipping our components with both, build and runtime type checking.

Reply:
ah, cool! thanks for the explanation!